### DISCUSSION TITLE
How to add build script in package.json ?

### DISCUSSION THREAD

**Discussion by xkeshav** [honojs/hono#1802] [General]
using Hono first time

below is my package.json

```json
{
  "scripts": {
    "dev": "tsx watch src/index.tsx"
  },
  "dependencies": {
    "@hono/node-server": "^1.3.1",
    "hono": "^3.11.2"
  },
  "devDependencies": {
    "tsx": "^3.12.2"
  }
}
```

which runs on local, now I want to deploy it in AWS amplify which check for build script and deploy

but unable to understand how and what to do for that?

someone can guide me.  

NOTe: cloudfare is not free and I really dont use that so prefer AWS amplify which does the job easy.

**Comment by linhub15** [comment]
Since you are using TypeScript you will need to use a bundler to compile it into JavaScript. 
- [esbuild](https://esbuild.github.io/) or Rollup.js, Webpack etc

Install it as a dev dependency and add the build script to your `package.json`
- https://esbuild.github.io/getting-started/#build-scripts

Then you can follow the Amplify documentation for "App build and test settings"
https://aws.amazon.com/getting-started/guides/deploy-webapp-amplify/module-three/

**Comment by EdamAme-x** [comment]
you should run `npm i -D esbuild`

and (if use esm)

```tsx
// build.ts
import esbuild from 'esbuild';

await esbuild.build({
  logLevel: 'info',
  entryPoints: ['./src/index.ts'], 
  outdir: './dist',
  outExtension: 
    '.js': '.mjs'
  }, // dist ext
  minify: true,
  bundle: true,
  platform: 'node',
  format: 'esm',
  banner: {
    js: 'import { createRequire as topLevelCreateRequire } from "module"; import url from "url"; const require = topLevelCreateRequire(import.meta.url); const __filename = url.fileURLToPath(import.meta.url); const __dirname = url.fileURLToPath(new URL(".", import.meta.url));',
  },
})
```

```json
"scripts": {
    "build": "tsx ./build.ts"
}
```
